screens/account/EnsureAuth.tsx

```diff
@@ -27,24 +29,25 @@ const Tabs = createMaterialBottomTabNavigator<RootTabParamList>();
  */
 const EnsureAuth = () => {
   const [initializing, setInitializing] = useState<boolean>(true);
-  const [signedIn, setSignedIn] = useState<boolean>(false);
+  const [signedIn, setSignedIn] = useRecoilState(signedInState);
   const [isEmailVerified, setIsEmailVerified] = useState<boolean>(
     auth.currentUser?.emailVerified ?? false
   );
 
-  const updateUserStatus = (user: any) => {
-    if (user !== null) {
-      setSignedIn(true);
-      setIsEmailVerified(user.emailVerified);
-    } else {
-      setSignedIn(false);
-      setIsEmailVerified(false);
-    }
-    setInitializing(false);
-  };
-
   // Listen to firebase authentication changes
-  useEffect(() => auth.onAuthStateChanged(updateUserStatus), []);
+  useEffect(() => {
+    const updateUserStatus = (user: any) => {
+      if (user !== null) {
+        setSignedIn(true);
+        setIsEmailVerified(user.emailVerified);
+      } else {
+        setSignedIn(false);
+        setIsEmailVerified(false);
+      }
+      setInitializing(false);
+    };
+    auth.onAuthStateChanged(updateUserStatus);
```

Comment:
When called, this function returns an unsubscriber.

To reduce memory leaks we should return that function as a cleanup for the useEffect.

```suggestion
return auth.onAuthStateChanged(updateUserStatus);
```